runtime/src/iree/hal/drivers/hip/hip_allocator.c

```diff
@@ -516,6 +528,19 @@ static iree_status_t iree_hal_hip_allocator_import_buffer(
   void* host_ptr = NULL;
   hipDeviceptr_t device_ptr = NULL;
 
+  int device_ordinal = 0;
+  if (params->queue_affinity) {
+    device_ordinal = iree_math_count_trailing_zeros_u64(params->queue_affinity);
+  }
+
+  status = IREE_HIP_CALL_TO_STATUS(
+      allocator->symbols,
+      hipCtxPushCurrent(
+          allocator->topology->devices[device_ordinal].hip_context));
+  if (!iree_status_is_ok(status)) {
+    return status;
+  }
+
   switch (external_buffer->type) {
```

Comment:
you can return-if-error first as part of validation and then keep the locals close to where they are initialized (in the switch statement).

```suggestion
  int device_ordinal = 0;
  if (params->queue_affinity) {
    device_ordinal = iree_math_count_trailing_zeros_u64(params->queue_affinity);
  }
  IREE_RETURN_IF_ERROR(IREE_HIP_CALL_TO_STATUS(
      allocator->symbols,
      hipCtxPushCurrent(
          allocator->topology->devices[device_ordinal].hip_context)));

  iree_status_t status = iree_ok_status();
  iree_hal_hip_buffer_type_t buffer_type = IREE_HAL_HIP_BUFFER_TYPE_DEVICE;
  void* host_ptr = NULL;
  hipDeviceptr_t device_ptr = NULL;
  switch (external_buffer->type) {
```